jax/experimental/jax2tf/jax2tf.py

```diff
@@ -213,7 +213,9 @@ def convert(fun_jax: Callable,
             polymorphic_shapes=None,
             with_gradient=True,
             enable_xla=True,
-            experimental_native_lowering="default") -> Callable:
+            experimental_native_lowering="default",
+            experimental_native_lowering_platforms=(),
+            experimental_native_lowering_strict_checks=True) -> Callable:
```

Comment:
You mentioned this in your description: `This will be easier to do when we clean up the code deprecated by jax.Array and pjit==jit APIs.`

Why is that the case? You can just allow this new behavior only under jax.Array (via jax.config.jax_array) flag and maybe do a proper integration? You can just error out if jax.Array is not enabled for new things. 

So if this leads to a better integration, then let's do that!

Reply:
See the comment above.